Source/Moya.swift

```diff
@@ -285,3 +287,18 @@ private struct CancellableWrapper: Cancellable {
 
 /// Make the Alamofire Request type conform to our type, to prevent leaking Alamofire to plugins.
 extension Request: RequestType { }
+
+/// Make `NSURLRequest` conform to `RequestType` so that it can be sent with `PluginType` method `willSendRequest(_:target:)`.
+extension NSURLRequest: RequestType {
+    public var request: NSURLRequest? {
+      return self
+    }
+
+    public func authenticate(usingCredential credential: NSURLCredential) -> Self {
```

Comment:
This approach is a little bit weird and will likely break the ```CredentialsPlugin```.

Personally I suppose we can just set ```startRequestsImmediately``` to ```false``` and add ```alamoRequest.resume()``` right behind where we created ```alamoRequest```